### YamlMime:ManagedReference
items:
- uid: System.ApplicationException
  id: ApplicationException
  children:
  - System.ApplicationException.#ctor
  - System.ApplicationException.#ctor(System.String)
  - System.ApplicationException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  - System.ApplicationException.#ctor(System.String,System.Exception)
  langs:
  - csharp
  name: ApplicationException
  nameWithType: ApplicationException
  fullName: System.ApplicationException
  type: Class
  summary: "Serve como a classe base para exceções definidas pelo aplicativo."
  remarks: >-
    > [!IMPORTANT]

    >  You should derive custom exceptions from the <xref:System.Exception> class rather than the ApplicationException class.  You should not throw an ApplicationException exception in your code, and you should not catch an ApplicationException exception unless you intend to re-throw the original exception.
  syntax:
    content: >-
      [System.Runtime.InteropServices.ComVisible(true)]

      public class ApplicationException : Exception
  inheritance:
  - System.Exception
  implements: []
  inheritedMembers: []
  platform:
  - net462
- uid: System.ApplicationException.#ctor
  id: '#ctor'
  parent: System.ApplicationException
  langs:
  - csharp
  name: ApplicationException()
  nameWithType: ApplicationException.ApplicationException()
  fullName: System.ApplicationException.ApplicationException()
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System
  summary: "Inicializa uma nova instância do <xref href=&quot;System.ApplicationException&quot;> </xref> classe."
  remarks: "Este construtor inicializa a <xref:System.Exception.Message%2A>propriedade da nova instância para uma mensagem fornecida pelo sistema que descreve o erro, como &quot;aplicativo erro.&quot;</xref:System.Exception.Message%2A> Esta mensagem leva em conta a cultura atual do sistema.       A tabela a seguir mostra os valores de propriedade inicial para uma instância de <xref:System.ApplicationException>.</xref:System.ApplicationException>      | Propriedade | Valor |   |--------------|-----------|   | <xref:System.Exception.InnerException%2A>| Uma referência nula (`Nothing` no Visual Basic). |   | <xref:System.Exception.Message%2A>| A cadeia de caracteres de mensagem de erro localizada. |</xref:System.Exception.Message%2A> </xref:System.Exception.InnerException%2A>"
  syntax:
    content: public ApplicationException ();
    parameters: []
  overload: System.ApplicationException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.ApplicationException.#ctor(System.String)
  id: '#ctor(System.String)'
  parent: System.ApplicationException
  langs:
  - csharp
  name: ApplicationException(String)
  nameWithType: ApplicationException.ApplicationException(String)
  fullName: System.ApplicationException.ApplicationException(String)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System
  summary: "Inicializa uma nova instância do <xref href=&quot;System.ApplicationException&quot;> </xref> classe com uma mensagem de erro especificada."
  remarks: "O conteúdo a `message` parâmetro se destina a ser compreendidos por humanos. O chamador desse construtor é necessário para garantir que essa cadeia de caracteres foi localizada para a cultura atual do sistema.       Esta mensagem leva em conta a cultura atual do sistema.       A tabela a seguir mostra os valores de propriedade inicial para uma instância de <xref:System.ApplicationException>.</xref:System.ApplicationException>      | Propriedade | Valor |   |--------------|-----------|   | <xref:System.Exception.InnerException%2A>| Uma referência nula (`Nothing` no Visual Basic). |   | `Message`| A cadeia de caracteres de mensagem de erro. |</xref:System.Exception.InnerException%2A>"
  syntax:
    content: public ApplicationException (string message);
    parameters:
    - id: message
      type: System.String
      description: "Uma mensagem que descreve o erro."
  overload: System.ApplicationException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.ApplicationException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  id: '#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)'
  parent: System.ApplicationException
  langs:
  - csharp
  name: ApplicationException(SerializationInfo,StreamingContext)
  nameWithType: ApplicationException.ApplicationException(SerializationInfo,StreamingContext)
  fullName: System.ApplicationException.ApplicationException(SerializationInfo,StreamingContext)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System
  summary: "Inicializa uma nova instância do <xref href=&quot;System.ApplicationException&quot;> </xref> classe com os dados serializados."
  remarks: "Este construtor é chamado durante a desserialização para reconstituir o objeto de exceção transmitido em um fluxo. Para obter mais informações, consulte [XML e serialização SOAP](~/add/includes/ajax-current-ext-md.md)."
  syntax:
    content: protected ApplicationException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);
    parameters:
    - id: info
      type: System.Runtime.Serialization.SerializationInfo
      description: "O objeto que contém os dados do objeto serializado."
    - id: context
      type: System.Runtime.Serialization.StreamingContext
      description: "As informações contextuais sobre a origem ou destino."
  overload: System.ApplicationException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.ApplicationException.#ctor(System.String,System.Exception)
  id: '#ctor(System.String,System.Exception)'
  parent: System.ApplicationException
  langs:
  - csharp
  name: ApplicationException(String,Exception)
  nameWithType: ApplicationException.ApplicationException(String,Exception)
  fullName: System.ApplicationException.ApplicationException(String,Exception)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System
  summary: "Inicializa uma nova instância do <xref href=&quot;System.ApplicationException&quot;> </xref> classe com uma mensagem de erro especificada e uma referência à exceção interna que é a causa dessa exceção."
  remarks: "O conteúdo a `message` parâmetro se destina a ser compreendidos por humanos. O chamador desse construtor é necessário para garantir que essa cadeia de caracteres foi localizada para a cultura atual do sistema.       Uma exceção que é lançada como um resultado direto de uma exceção anterior deve incluir uma referência à exceção anterior no <xref:System.Exception.InnerException%2A>propriedade.</xref:System.Exception.InnerException%2A> O <xref:System.Exception.InnerException%2A>propriedade retorna o mesmo valor que é passado para o construtor, ou uma referência nula se o <xref:System.Exception.InnerException%2A>propriedade não fornecer o valor da exceção interna para o construtor.</xref:System.Exception.InnerException%2A> </xref:System.Exception.InnerException%2A>       A tabela a seguir mostra os valores de propriedade inicial para uma instância de <xref:System.ApplicationException>.</xref:System.ApplicationException>      | Propriedade | Valor |   |--------------|-----------|   | <xref:System.Exception.InnerException%2A>| A referência à exceção interna. |   | <xref:System.Exception.Message%2A>| A cadeia de caracteres de mensagem de erro. |</xref:System.Exception.Message%2A> </xref:System.Exception.InnerException%2A>"
  syntax:
    content: public ApplicationException (string message, Exception innerException);
    parameters:
    - id: message
      type: System.String
      description: "A mensagem de erro que explica o motivo da exceção."
    - id: innerException
      type: System.Exception
      description: "A exceção que é a causa da exceção atual. Se o `innerException` parâmetro não é uma referência nula, a exceção atual é gerada em um <xref uid=&quot;langword_csharp_catch&quot; name=&quot;catch&quot; href=&quot;&quot;> </xref> bloco que manipula a exceção interna."
  overload: System.ApplicationException.#ctor*
  exceptions: []
  platform:
  - net462
references:
- uid: System.Exception
  isExternal: false
  name: System.Exception
- uid: System.ApplicationException.#ctor
  parent: System.ApplicationException
  isExternal: false
  name: ApplicationException()
  nameWithType: ApplicationException.ApplicationException()
  fullName: System.ApplicationException.ApplicationException()
- uid: System.ApplicationException.#ctor(System.String)
  parent: System.ApplicationException
  isExternal: false
  name: ApplicationException(String)
  nameWithType: ApplicationException.ApplicationException(String)
  fullName: System.ApplicationException.ApplicationException(String)
- uid: System.String
  parent: System
  isExternal: true
  name: String
  nameWithType: String
  fullName: System.String
- uid: System.ApplicationException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  parent: System.ApplicationException
  isExternal: false
  name: ApplicationException(SerializationInfo,StreamingContext)
  nameWithType: ApplicationException.ApplicationException(SerializationInfo,StreamingContext)
  fullName: System.ApplicationException.ApplicationException(SerializationInfo,StreamingContext)
- uid: System.Runtime.Serialization.SerializationInfo
  parent: System.Runtime.Serialization
  isExternal: false
  name: SerializationInfo
  nameWithType: SerializationInfo
  fullName: System.Runtime.Serialization.SerializationInfo
- uid: System.Runtime.Serialization.StreamingContext
  parent: System.Runtime.Serialization
  isExternal: true
  name: StreamingContext
  nameWithType: StreamingContext
  fullName: System.Runtime.Serialization.StreamingContext
- uid: System.ApplicationException.#ctor(System.String,System.Exception)
  parent: System.ApplicationException
  isExternal: false
  name: ApplicationException(String,Exception)
  nameWithType: ApplicationException.ApplicationException(String,Exception)
  fullName: System.ApplicationException.ApplicationException(String,Exception)
- uid: System.ApplicationException.#ctor*
  parent: System.ApplicationException
  isExternal: false
  name: ApplicationException
  nameWithType: ApplicationException.ApplicationException
